### PR TITLE
REST authentication: Remove exception class name from failure message

### PR DESCRIPTION
**Pull request by johnjcool**
our security review check all error views. 
```
server.error.include-stacktrace=NEVER
```
deactivating stacktrace did not filter out messages with cas internal information. so you get information about the underlying technology and look how you can hack the whole system. 

What sensitive information are you referring to?
CAS Class Information

What framework are you referring to?
CAS Rest Auth

If you're using REST-based authentication and your API is passing along sensitive information as part of a failure, have you considered to change the API to not do that?
CAS is passing this info to client if something goes wrong. A hacker provoke errors to get such information. 

Are there test cases accounting for the changes you have made?
There are no tests for this class and i need an initial test class, because i do not know your whole test infrastructure. 

Does this patch need to be ported forward to master?
yes

### CODE REVIEW

**Comment by johnjcool** [comment]
I am disappointed to get no feedback from any cas core developer... whether on github nor on Gitter. The build problem are LDAP tests. Not my fault. The branch is broken. Did you really want help to fix bugs? Think about your communication strategy.

**Comment by mmoayyed** [comment]
Can you please review:

1. What sensitive information are you referring to?
2. What framework are you referring to? 
3. If you're using REST-based authentication and your API is passing along sensitive information as part of a failure, have you considered to change the API to not do that? 
4. Are there test cases accounting for the changes you have made?
5. Does this patch need to be ported forward to master?

**Comment by mmoayyed** [comment]
> deactivating stacktrace did not filter out messages with cas internal information. 

What internal information? can you please provide an example?

> What sensitive information are you referring to?
> CAS Class Information

Can you provide an example?

**Comment by mmoayyed** [comment]
> look how you can hack the whole system.

Looking at the patch, I see that the exception class name is being removed. I think your statement of "hacker can look at this and hack the system" is somewhat exaggerated but you do have a valid point that this sort of thing should not preferably leak out.

File: support/cas-server-support-rest-core/src/main/java/org/apereo/cas/support/rest/resources/RestResourceUtils.java [hunk 1]
```diff
@@ -39,8 +39,12 @@
             .findAndRegisterModules()
             .configure(DeserializationFeature.ADJUST_DATES_TO_CONTEXT_TIME_ZONE, false)
             .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
-            .setSerializationInclusion(JsonInclude.Include.NON_EMPTY)
-            .enableDefaultTyping(ObjectMapper.DefaultTyping.NON_FINAL, JsonTypeInfo.As.PROPERTY);
+            .setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
+        MAPPER.activateDefaultTyping(
+                MAPPER.getPolymorphicTypeValidator(),
+                ObjectMapper.DefaultTyping.NON_FINAL,
+                JsonTypeInfo.As.PROPERTY
```
Comment (mmoayyed) [inline support/cas-server-support-rest-core/src/main/java/org/apereo/cas/support/rest/resources/RestResourceUtils.java:46]: Why this change?

File: support/cas-server-support-rest/src/test/java/org/apereo/cas/support/rest/RestResourceUtilsTests.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import static org.mockito.Mockito.mock;
+
+/**
+ * This is {@link RestResourceUtilsTests}.
+ *
+ * @author john.j.cool
+ * @since 6.1.6
+ */
+@ExtendWith(MockitoExtension.class)
+public class RestResourceUtilsTests {
```
Comment (mmoayyed) [inline support/cas-server-support-rest/src/test/java/org/apereo/cas/support/rest/RestResourceUtilsTests.java:33]: Tag with `@Tag("RestfulApi")`

File: support/cas-server-support-rest/src/test/java/org/apereo/cas/support/rest/RestResourceUtilsTests.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    @Test
+    public void verifyNoFrameworkClassSendToClient() {
+        val request = mock(HttpServletRequest.class);
+        val msg = "test";
+        val map = CollectionUtils.<String, Throwable>wrap(
+                UnauthorizedAuthenticationException.class.getSimpleName(),
+                new UnauthorizedAuthenticationException(msg)
+        );
+
+        ResponseEntity<String> response = RestResourceUtils.createResponseEntityForAuthnFailure(
```
Comment (mmoayyed) [inline support/cas-server-support-rest/src/test/java/org/apereo/cas/support/rest/RestResourceUtilsTests.java:54]:
```suggestion
        val response = RestResourceUtils.createResponseEntityForAuthnFailure(
```

**Comment by mmoayyed** [comment]
I updated the title to reflect the intention of this patch accurately. Whenever you are done with the changes here, please post a PR to the master branch as well. Thanks again!

File: support/cas-server-support-rest-core/src/main/java/org/apereo/cas/support/rest/resources/RestResourceUtils.java (hunk 1 above)
Comment (johnjcool) [reply support/cas-server-support-rest-core/src/main/java/org/apereo/cas/support/rest/resources/RestResourceUtils.java:46]: enableDefaultTyping is deprecated and sonar check points to this https://rules.sonarsource.com/java/RSPEC-4544

Comment (johnjcool) [reply support/cas-server-support-rest-core/src/main/java/org/apereo/cas/support/rest/resources/RestResourceUtils.java:46]: under the hood jackson is doing the same in enableDefaultTyping.

Comment (mmoayyed) [reply support/cas-server-support-rest-core/src/main/java/org/apereo/cas/support/rest/resources/RestResourceUtils.java:46]: Indeed. However, since that is unrelated to your changeset, please reset back to original. If you want to clean up deprecated instances such as this, that would be a separate PR and one that needs to be applied everywhere to be consistent.

**Comment by mmoayyed** [comment]
This failure seems relevant:

```
org.apereo.cas.support.rest.ServiceTicketResourceTests > creationOfSTWithAuthenticationException() FAILED
    org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
        at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:40)
        at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:35)
        at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:162)
        at org.apereo.cas.support.rest.ServiceTicketResourceTests.creationOfSTWithAuthenticationException(ServiceTicketResourceTests.java:164)
```

**Comment by johnjcool** [comment]
it doesn't seem that these test errors have something to do with the change.

**Comment by mmoayyed** [comment]
This seems related:

```
org.apereo.cas.support.rest.TicketGrantingTicketResourceTests > creationOfTGTWithAuthenticationException() FAILED
    org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
        at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:40)
        at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:35)
        at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:162)
        at org.apereo.cas.support.rest.TicketGrantingTicketResourceTests.creationOfTGTWithAuthenticationException(TicketGrantingTicketResourceTests.java:167)
```

**Comment by johnjcool** [comment]
thanks. fixed.
